DEVELOPMENT.md

```diff
@@ -0,0 +1,198 @@
+- [bugsnag-plugin-android-ndk](bugsnag-plugin-android-ndk/README.md), responsible for capturing NDK errors
+- [bugsnag-plugin-android-anr](bugsnag-plugin-android-anr/README.md), responsible for capturing ANRs
+
+[Gradle tasks](https://docs.gradle.org/current/userguide/tutorial_using_tasks.html) can generally be run on the entire project, or on an individual module: 
+
+```shell
+./gradlew build // builds whole project
+./gradlew bugsnag-plugin-android-anr:build // builds one module only
+```
+
+# Static analysis
+
+Several static analysis checks are run against bugsnag-android to maintain the quality of the codebase. `./gradlew lint ktlintCheck detekt checkstyle` runs them all at once.
+
+## Android Lint
+
+[Android Lint](https://developer.android.com/studio/write/lint) Runs Android-specific static analysis checks. Warnings can be suppressed by following [this guide](https://developer.android.com/studio/write/lint#config).
+
+```shell
+./gradlew lint
+```
+
+## Ktlint
+
+[Ktlint](https://github.com/pinterest/ktlint) runs style checks on Kotlin code.
+
+```shell
+./gradlew ktlintCheck
+```
+
+Some violations (but not all) can be automatically addressed by running the following task:
+
+```shell
+./gradlew ktlintFormat
+```
+
+## Detekt
+
+[Detekt](https://github.com/detekt/detekt) runs static analysis checks on Kotlin code.
+
+```shell
+./gradlew detekt
+```
+
+Warnings can be suppressed by running the following task:
+
+```shell
+./gradlew detektBaseline
+```
+
+## Checkstyle
+
+[Checkstyle](https://github.com/checkstyle/checkstyle) runs style checks on Java code.
+
+```shell
+./gradlew checkstyle
+```
+
+# Unit tests
+
+Unit tests are implemented using [Junit](https://developer.android.com/training/testing/unit-testing/local-unit-tests) and can be run with the following:
+
+`./gradlew test`
+
+Unit tests run on the local JVM and cannot access Android OS classes.
+
+# Instrumentation tests
+
+Instrumentation tests are implemented using [Junit](https://developer.android.com/training/testing/unit-testing/instrumented-unit-tests) and can be run with the following:
+
+`./gradlew connectedCheck`
+
+Instrumentation tests require an Android emulator or device to run, and they can access Android OS classes.
+
+# Example app
+
+The example app can be found in [examples/sdk-app-example](examples/sdk-app-example). It contains example code which triggers crashes.
+
+You should open `examples/sdk-app-example` as a separate Android Studio project and [run the app](https://developer.android.com/training/basics/firstapp/running-app). This will require a connected device or emulator.
+
+The app **does not use the local development version of Bugsnag**. To test local development changes, you will need to build the library locally.
+
+## Building the Library locally
+
+The following command assembles bugsnag-android for local use:
+
+```shell
+./gradlew assembleRelease publishToMavenLocal -PVERSION_NAME=9.9.9
```

Comment:
Does setting `VERSION_NAME` replace the two steps of modifying `gradle.properties` and `sdk-app-example/build.gradle` to have the same large, unassigned bugsnag-android version numbers?

Reply:
Updated to state that the example app's version number must be updated too